crates/ruff_python_formatter/src/expression/number.rs

```diff
@@ -1,24 +1,24 @@
 use std::borrow::Cow;
 
-use ruff_python_ast::ExprConstant;
+use ruff_python_ast::ExprNumberLiteral;
 use ruff_text_size::{Ranged, TextSize};
 
 use crate::prelude::*;
 
 pub(super) struct FormatInt<'a> {
-    constant: &'a ExprConstant,
+    number: &'a ExprNumberLiteral,
 }
 
 impl<'a> FormatInt<'a> {
-    pub(super) fn new(constant: &'a ExprConstant) -> Self {
-        debug_assert!(constant.value.is_int());
-        Self { constant }
+    pub(super) fn new(number: &'a ExprNumberLiteral) -> Self {
+        debug_assert!(number.value.is_int());
```

Comment:
I wonder if it still makes sense to have these 3 structs or if we should just inline it into `FormatNumber` and use one big match.